lms/static/scripts/frontend_apps/components/BasicLtiLaunchApp.js

```diff
@@ -209,31 +249,30 @@ export default function BasicLtiLaunchApp() {
     } else {
       return iFrame;
     }
-  }
-
-  return (
-    <Fragment>
-      {ltiLaunchState.state === 'fetching-url' && (
-        <Spinner className="BasicLtiLaunchApp__spinner" />
-      )}
-      {ltiLaunchState.state === 'authorizing' && (
-        <Dialog
-          title="Authorize Hypothesis"
-          role="alertdialog"
-          buttons={[
-            <Button
-              onClick={authorizeAndFetchUrl}
-              className="BasicLtiLaunchApp__button"
-              label="Authorize"
-              key="authorize"
-            />,
-          ]}
-        >
-          <p>Hypothesis needs your authorization to launch this assignment.</p>
-        </Dialog>
-      )}
-      {ltiLaunchState.state === 'error' &&
-        ltiLaunchState.failedAction === 'fetch-url' && (
+  } else {
+    // Render either a Spinner or Dialog.
+    return (
+      <Fragment>
+        {showSpinner() && <Spinner className="BasicLtiLaunchApp__spinner" />}
+        {errorState === 'error-authorizing' && (
+          <Dialog
```

Comment:
It might be useful to extract a helper component out of these dialogs to ensure consistency for the parts that are supposed to be the same across all of them.

Reply:
Looked closely, but I'm gonna leave that for another pass, each dialog is slightly different than the other and I have another ticket to refactor the Dialog's css so it may be a good time to revisit this for readability when doing that ticket.

Reply:
OK, that's fine by me.